src/components/AddOn/core/AddOnRegister/AddOnRegister.tsx

```diff
@@ -1,158 +1,166 @@
-
-  const [formState, setFormState] = useState<formStateTypes>({
-    pluginName: '',
-    pluginCreatedBy: '',
-    pluginDesc: '',
-    pluginInstallStatus: false,
-    installedOrgs: [currentUrl],
-  });
-
-  const handleRegister = async () => {
-    console.log(formState);
-    const { data } = await create({
-      variables: {
-        $pluginName: formState.pluginName,
-        $pluginCreatedBy: formState.pluginCreatedBy,
-        $pluginDesc: formState.pluginDesc,
-        $pluginInstallStatus: formState.pluginInstallStatus,
-        $installedOrgs: formState.installedOrgs,
-      },
-    });
-
-    if (data) {
-      toast.success('Plugin Added Successfully');
-      setTimeout(() => {
-        window.location.reload();
-      }, 2000);
-    }
-    console.log('Data is ', data);
-  };
-  return (
-    <>
-      <Button
-        className={styles.modalbtn}
-        variant="primary"
-        onClick={handleShow}
-      >
-        <i className="fa fa-plus"></i>
-        {t('addNew')}
-      </Button>
-
-      <Modal show={show} onHide={handleClose}>
-        <Modal.Header closeButton>
-          <Modal.Title> {t('addPlugin')}</Modal.Title>
-        </Modal.Header>
-        <Modal.Body>
-          <Form>
-            <Form.Group className="mb-3" controlId="registerForm.PluginName">
-              <Form.Label>{t('pluginName')}</Form.Label>
-              <Form.Control
-                type="text"
-                placeholder={t('pName')}
-                autoComplete="off"
-                required
-                value={formState.pluginName}
-                onChange={(e) => {
-                  setFormState({
-                    ...formState,
-                    pluginName: e.target.value,
-                  });
-                }}
-              />
-            </Form.Group>
-            <Form.Group className="mb-3" controlId="registerForm.PluginName">
-              <Form.Label>{t('creatorName')}</Form.Label>
-              <Form.Control
-                type="text"
-                placeholder={t('cName')}
-                autoComplete="off"
-                required
-                value={formState.pluginCreatedBy}
-                onChange={(e) => {
-                  setFormState({
-                    ...formState,
-                    pluginCreatedBy: e.target.value,
-                  });
-                }}
-              />
-            </Form.Group>
-            <Form.Group className="mb-3" controlId="registerForm.PluginURL">
-              <Form.Label>{t('pluginDesc')}</Form.Label>
-              <Form.Control
-                // type="text"
-                rows={3}
-                as="textarea"
-                placeholder={t('pDesc')}
-                required
-                value={formState.pluginDesc}
-                onChange={(e) => {
-                  setFormState({
-                    ...formState,
-                    pluginDesc: e.target.value,
-                  });
```

Comment:
hello @anshgoyalevil, I'd like to know why have you commented on this code and what tests were failing for this. the test for this only contains some UI checks to make sure proper UI is rendered for the plugin. Commenting on this code would remove that UI from admin.
Also  how this helps to solve #819 as it was just intended to implement a check that tests should fail when there's a warning in the console for that test case, right?

Reply:
@SiddheshKukade 

Thanks for asking out.
This code was for adding the plugin, and the plugins system doesn't work. With this code in place, the coverage would fall.

I have a question too. Should we test a component, if it doesn't work? Are we aiming for coverage or for features?

Moreover, if you think, there was no problem with the code, since you are also having some valid points. I would add the code back. But, I don't think the implementation is correct. Plugins are not posts. They are predefined, and not taken as a user input.